packages/back-end/src/types/Integration.ts

```diff
@@ -157,6 +157,57 @@ export interface TestQueryResult {
   duration: number;
 }
 
+export interface RawInformationSchema {
+  table_name: string;
+  table_catalog: string;
+  table_schema: string;
+  column_count: string;
+}
+
+export interface Column {
+  columnName: string;
+  path?: string;
+  dataType: string;
+}
+
+export interface Table {
+  tableName: string;
+  path: string;
+  id: string;
+  numberOfColumns: number;
+}
+
+export interface Schema {
+  schemaName: string;
+  tables: Table[];
+  path?: string;
+}
+
+export interface InformationSchema {
+  databaseName: string;
+  path?: string;
+  schemas: Schema[];
+}
+
+export interface InformationSchemaInterface {
+  id: string;
+  databases: InformationSchema[];
+  dateCreated: Date;
+  dateUpdated: Date;
+  organization: string;
+}
+
+export interface InformationSchemaTablesInterface {
+  id: string;
+  organization: string;
+  tableName: string;
+  tableSchema: string;
+  databaseName: string;
+  columns: Column[];
+  dateCreated: Date;
+  dateUpdated: Date;
+}
```

Comment:
I would add a data source id to both of these interfaces. I'm thinking of the case when a data source is deleted, finding all of these documents to delete is going to be a giant pain right now.

Reply:
Also created a ticket on the Linear project to delete the related informationSchemas when a datasource is deleted.